### ISSUE TITLE
[ProfileShowcase] Add network preferences support for accounts

### ISSUE THREAD

**Issue by alexjba** [status-im/status-desktop#14226]
### Description

The purpose of this task is to provide the backend support for network preferences in the showcase accounts. The network preferences should be part of the showcase `save` API and showcase for a contact models.

When changing the network preferences from wallet settings the profile showcase should be updated.

TBD: the network preferences format

              Here we discussed we may need the **network preferences** too. Not sure which was the final decisions here @alexjba ?

_Originally posted by @noeliaSD in https://github.com/status-im/status-desktop/pull/13998#discussion_r1542906129_

**Comment by MishkaRogachev** [comment]
A few questions:
1. Are network preferences = chain Ids?
2. Are network preferences same for all accounts? (there may be logic issues if so)
3. If a user dispatches the profile showcase, should network preferences depend on test net mode? 
4. If a user receives a profile showcase for a contact, should network preferences depend test net mode? Sender or receiver side?
5. There is already **chain_Id** field for collectibles and unverified tokens. Should we filter these entries by network preferences?

**Comment by alexjba** [comment]
> A few questions:
> 
> 1. Are network preferences = chain Ids?
Not necessarily, but could be. In qml we interact mostly with chain short names.

> 2. Are network preferences same for all accounts? (there may be logic issues if so)
No. You can setup any chain ids in wallet settings. It's per account.

<img width="1094" alt="Screenshot 2024-03-29 at 17 10 51" src="https://github.com/status-im/status-desktop/assets/47811206/030be9f2-eca6-47f9-a7ce-adaf3a063aed">

> 3. If a user dispatches the profile showcase, should network preferences depend on test net mode?
> 4. If a user receives a profile showcase for a contact, should network preferences depend test net mode? Sender or receiver side?

Not sure, but I don't see any relation between network preferences and test net mode. I see is as a preference saying with what chains the user would like to interact.
Maybe @alaibe knows more?

> 5. There is already **chain_Id** field for collectibles and unverified tokens. Should we filter these entries by network preferences?

No, just display the data to the user. Sender (settings and profile preview) and receiver side

**Comment by noeliaSD** [comment]
@MishkaRogachev do you know if it will be in for milestone 2.29.0? @alexjba does the **UI team** need a task linked to this?

**Comment by noeliaSD** [comment]
> I think it would be wise to do it in same iteration with assets, in case network preferences should be used to querry assets and collectibles? Or it just for Ui for address sharing?

I can be wrong but I think it's only for sharing wallet account network preferences.

**Comment by alexjba** [comment]
> @MishkaRogachev do you know if it will be in for milestone 2.29.0? @alexjba does the **UI team** need a task linked to this?

There is a linked one: https://github.com/status-im/status-desktop/issues/14227

**Comment by noeliaSD** [comment]
> > @MishkaRogachev do you know if it will be in for milestone 2.29.0? @alexjba does the **UI team** need a task linked to this?
> 
> There is a linked one: #14227

I'd say that if _messenger team_ can work on it during milestone **2.29.0** we should also considere it for the same one bc one of our goals is finishing profile showcase pending tasks!

**Comment by noeliaSD** [comment]
> > Not sure, but I don't see any relation between network preferences and test net mode.
> 
> chain Ids are different for testnet. If it's ui-only thing we can store network preferences as strings

I've asked the wallet team (cc @Khushboo-dev-cpp ) and my understanding here is that, if we are using as the base model, a wallet model, these chain ids changes from testnet/mainnet are directly updated in the wallet internals so the model exposed to the UI will be ready to display the correct values.

**Comment by noeliaSD** [comment]
> > > @MishkaRogachev do you know if it will be in for milestone 2.29.0? @alexjba does the **UI team** need a task linked to this?
> > 
> > 
> > There is a linked one: #14227
> 
> I'd say that if _messenger team_ can work on it during milestone **2.29.0** we should also considere it for the same one bc one of our goals is finishing **profile showcase** pending tasks!

cc: @jrainville is there space to include this task on 2.29.0?

**Comment by jrainville** [comment]
Considering this 
> I think it would be wise to do it in same iteration with assets, in case network preferences should be used to querry assets and collectibles? Or it just for Ui for address sharing?

Then yeah, we'll try to get it in 2.29

**Comment by jrainville** [comment]
Moving Profile showcase issues to 2.30 for now as it's no longer priority
